communis-mission/src/main/java/com/rhema/communis/mission/region/RegionService.java

```diff
@@ -30,16 +35,24 @@ public Region createSubRegion(String parentRegionId, Region subRegion) {
             throw new ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, "Region " + parentRegionId + " not found");
         }
         subRegion.setParentRegion(parentRegionId);
-        this.saveOrUpdate(subRegion);
+        this.save(subRegion);
 
         parent.getSubRegions().add(subRegion.getId());
-        this.saveOrUpdate(parent);
-        logger.info("Created new Sub Region with sub region name " + subRegion.getMissionRegionName());
+        this.update(parent);
+        logger.info("Created new Sub Region with sub region name " + subRegion.getName());
         return subRegion;
     }
 
     public List<Region> findSubRegions(String regionId) {
         Region parent = this.find(regionId);
         return this.regionRepository.findByIds(parent.getSubRegions());
     }
+
+    @Override
+    public void onBeforeConvert(BeforeConvertEvent<Region> event) {
```

Comment:
I think it should be either `onConvert` or `beforeConvert`

Reply:
@robinvk6 is overriding the function coming from parent class. We cannot change the parent class function as the parent class is - `AbstractMongoEventListener`, which is part of spring - `org.springframework.data.mongodb.core.mapping.event`